crates/core/src/nucleus/actions/run_validation_callback.rs

```diff
@@ -10,56 +10,57 @@ use holochain_core_types::error::HolochainError;
-    let call2 = call.clone();
-
-    let validation_result: ValidationResult = match ribosome::run_dna(
-        Some(call2.clone().parameters.to_bytes()),
-        WasmCallData::new_callback_call(cloned_context, call2),
-    ) {
-        Ok(call_result) => {
-            if call_result.is_null() {
-                Ok(())
-            } else {
-                Err(ValidationError::Fail(call_result.to_string()))
-            }
-        }
-        // TODO: have "not matching schema" be its own error
-        Err(HolochainError::RibosomeFailed(error_string)) => {
-            if error_string == "Argument deserialization failed" {
-                Err(ValidationError::Error(
-                    String::from("JSON object does not match entry schema").into(),
-                ))
-            } else {
-                // an unknown error from the ribosome should panic rather than
-                // silently failing validation
-                panic!(error_string)
-            }
-        }
-        Err(error) => panic!(error.to_string()), // same here
-    };
-
-    let metric_name = format!(
-        "run_validation_callback.{}.{}.latency",
+    let metric_name_prefix = format!(
+        "run_validation_callback.{}.{}",
         call.zome_name, call.fn_name
     );
-    let latency = clock.elapsed().unwrap().as_millis();
-    let metric = Metric::new(metric_name.as_str(), latency as f64);
-    context.metric_publisher.write().unwrap().publish(&metric);
 
-    validation_result
+    with_latency_publishing!(
+        metric_name_prefix,
+        context.metric_publisher,
+        |_address, call: CallbackFnCall, context: &Arc<Context>| {
+            let cloned_context = context.clone();
+
+            match ribosome::run_dna(
+                Some(call.clone().parameters.to_bytes()),
+                WasmCallData::new_callback_call(cloned_context, call),
+            ) {
+                Ok(call_result) => {
+                    if call_result.is_null() {
+                        Ok(())
+                    } else {
+                        Err(ValidationError::Fail(call_result.to_string()))
+                    }
+                }
+                // TODO: have "not matching schema" be its own error
+                Err(HolochainError::RibosomeFailed(error_string)) => {
+                    if error_string == "Argument deserialization failed" {
+                        Err(ValidationError::Error(
+                            String::from("JSON object does not match entry schema").into(),
+                        ))
+                    } else {
+                        // an unknown error from the ribosome should panic rather than
+                        // silently failing validation
+                        panic!(error_string)
+                    }
+                }
+                Err(error) => panic!(error.to_string()), // same here
+            }
+        },
+        address,
+        call,
+        context
```

Comment:
Sorry, not relevant to this PR, but I'm just seeing `with_latency_publishing!` in action for the first time so I have a question. @struktured what is the reason for supplying a closure that takes args and then passing them into the closure in the macro? Why not just have a zero arity closure and move the values needed into it?

Reply:
@maackle It's a good question. The idea behind the macro is you usually pass a defined function and its arguments to the macro and it invokes the function for you. In this case I didn't bother defining a separate function so it appears more awkward. Will take your advice and make the closure simpler.

Reply:
Simplified the closure to take just a unit arg. My macro would have to slightly change to support the zero argument case (can do in a future PR though).